### DISCUSSION TITLE
Re-enabling old games?

### DISCUSSION THREAD

**Discussion by krazyjakee** [kevincar/DPServer#17] [Q&A]
Does/will this project enable me to play old directplay games such as commandos 2 over LAN?
It's just, I've tried hamachi, zerotier and gameranger and all don't work. I believe it's because directplay has special requirements that are incompatible with those other "hole punching" solutions. Presumably this project will solve it?

**Comment by kevincar** [comment]
This project doesn't really focus on establishing game play over a Local Area Network (LAN); i.e, between machines running the applications (games) on the same home network for instance. This project assumes that the DirectPlay application in question is already capable of connecting with other applications over a home LAN. 

The purpose of this project is to make it easier to find possible connections (e.g., a hosted game session) *outside* of a LAN. The hope is that you can connect to a host session without needing to know the IP address of the host. To picture how this will work, imagine having 4 computers (A, B, C, and D). A is running the DirectPlayHelper server and B, C, and D are running a DirectPlay Application and they hope to connect with one another, but they're all on different LANs. B, C, and D use the DirectPlayHelper Client to connect to the server. Then, using an application that relies on DirectPlay (like Commandos 2), B can host a session, while C and D attempt to join. DirectPlay packets sent by C or D are intercepted by the DirectPlay client and sent to the DirectPlayServer on A. A then relays this message to all connected to computer to find potential sessions or games. Connections are initially establish by routing the packets through A. Holepunching is then attempted, but if it fails, the communications remain in relay mode, in which messages are sent between computers through A.

The project is pretty far from an initial beta release, but I anticipate that the ability to test basic connectivity between application should be available soon. 

Hamachi, Zerotier, and GameRanger are VPN / game services. This project is much much more focused on applications that only use DirectPlay for communications, which theoretically should work for an application like Commandos 2.

**Reply by krazyjakee** [reply]
Thanks for the detailed response, I really appreciate it. Exciting!
